src/pytorch_lightning/strategies/launchers/xla.py

```diff
@@ -139,30 +135,3 @@ def _collect_rank_zero_results(self, trainer: "pl.Trainer", results: Any) -> Opt
         self.add_to_queue(trainer, extra)
 
         return _WorkerOutput(best_model_path, weights_path, trainer.state, results, extra)
-
-
-def _save_spawn(
```

Comment:
So the conclusion was that this save_spawn was somehow the reason for the original pickling issue we discussed on this pr?

I'm fine with having the alternative of
`_rank_teardown(process_idx)`, but the benefit of the save_spawn function was that if you used it, it would guarantee you with the correct teardown and you don't forget to put special teardown logic in your own function.

Reply:
I believe it was. At least the error disappear after removing it.

I agree with your second point. I'm going to open a separate PR proposing to use the launchers in our tests to launch